trezorlib/tests/device_tests/test_msg_signtx_capricoin.py

```diff
@@ -0,0 +1,220 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2012-2018 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+import pytest
+
+from trezorlib import messages as proto
+from trezorlib.tests.device_tests.test_msg_signtx import check_sign_tx
+from trezorlib.tools import CallException, parse_path
+
+from .common import TrezorTest
+
+TXHASH_915340 = bytes.fromhex(
+    "915340ecc7466d287596f1f5b1fa0c1fa78c5b76ede0dff978fd6a1ca31eee24"
+)
+
+TXHASH_f8a9bb = bytes.fromhex(
+    "f8a9bb54de8295de3cd681d89b69834812eb5222d04b3fec206eb273f69ffdbb"
+)
+
+TXHASH_2e824c = bytes.fromhex(
+    "2e824ca52962912c120de0b57a763b4355727697f7ae589dc1caf526789fbf0f"
+)
+
+TXHASH_3d00cb = bytes.fromhex(
+    "3d00cb457a7a0d8f491296340696271b9440a4b50e5429cf5e51fe128bce10d8"
+)
+
+TXHASH_8c4553 = bytes.fromhex(
+    "8c4553a62d28a2aa605dc82e80c8e30fdadd49bf950d902c37c34428d5ff58a1"
+)
+
+
+@pytest.mark.xfail
+@pytest.mark.capricoin
+class TestMsgSigntx(TrezorTest):
```

Comment:
This is effectively testing the Bitcoin signing flow except with a timestamp. That's not needed. Please reduce this to one or at most two tests where you provide the timestamp and check that the resulting signature is correct, without involving `check_sign_tx`.

You could also do a fail-test, asserting that signing for Capricoin fails if timestamp is missing.
(Or test for other possible features that work *differently* for Capricoin.)

Reply:
So I should write my own super short/simple version of check_sign_tx, right? Any example I could steal from? If not don't worry I'll manage. (tomorrow/thursday)

Reply:
You're right a single test is sufficient. I was thinking about the fail-test. We could add asserts like below but there is probably not much value.
```python
        with self.client:
            signatures, tx = btc.sign_tx(
                self.client,
                "Capricoin",
                [inp1, inp2],
                [out1],
                version=1,
                timestamp=None,
            )

        assert timestamp_signatures != signatures
        assert len(tx) < len(timestamp_tx)

```
Waiting for your feedback.

Reply:
will this non-timestamped signature be _valid_ for Capricoin?

if yes, then this is a reasonable test.
if not, you should update trezor-core code to throw a failure, and test for that

Reply:
actually, if the timestamp is not mandatory, you should also assert that this creates an expected signature, not just that it's different (you're right about that; checking that the result is different is by itself not interesting)

Reply:
timestamp is mandatory so non-timestamped signature is invalid for Capricoin.

So overwintered (only for Zcash) is optional for Zcash and that's why it doesn't have validation in core / signing.py?

I'll work on new version, thanks.

Reply:
actually we had an internal discussion and let's keep it as it is now, without the fail-test.

it should be there technically, but it is not worth complicating the trezor-core implementation over it

Reply:
Agree with everything :). So I think it's your turn, thanks.